TypeCobol/Compiler/Symbols/RedefinesSymbol.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.IO;
+using TypeCobol.Compiler.Scopes;
+
+namespace TypeCobol.Compiler.Symbols
+{
+    /// <summary>
+    /// Represents a Redefines Symbol, for instance in the example Below it is the variable B.
+    /// ex:
+    /// 05  A PICTURE X(6).
+    /// 05  B REDEFINES A.
+    ///     10 B-1          PICTURE X(2).
+    ///     10 B-2          PICTURE 9(4).
+    /// 05  C PICTURE 99V99.
+    /// </summary>
+    public class RedefinesSymbol : VariableSymbol
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="name">Symbol 's name</param>
+        /// <param name="redefined">The redefined Symbol</param>
+        public RedefinesSymbol(string name, VariableSymbol redefined) : base(name)
+        {
+            base.SetFlag(Flags.Redefines, true);
+            Redefined = redefined;
+        }
+
+        /// <summary>
+        /// The redefined symbol
+        /// </summary>
+        public VariableSymbol Redefined
+        {
+            get;
+            internal set;
+        }
+
+        /// <summary>
+        /// Get the Top REDEFINES in case of REDEFINES OF REDEFINES suite.
+        /// </summary>
+        public VariableSymbol TopRedefined => Redefined != null && Redefined.HasFlag(Flags.Redefines) ? ((RedefinesSymbol) Redefined).TopRedefined : Redefined;
+
+        /// <summary>
+        /// When a RedefinesSymbol is normalized it  Redefined Symbol must change.
+        /// </summary>
+        /// <param name="domain"></param>
+        internal override void NormalizeExpandedSymbol(Domain<VariableSymbol> domain)
+        {
+            base.NormalizeExpandedSymbol(domain);
+            Container<VariableSymbol>.Entry entry = domain.Lookup(Redefined.Name);
```

Comment:
This is a not a check here. The constraint on Redefines are currently checked in the ProgramSymbolTableBuilder. Here we attach the expanded Redefined symbol to the expanded Redefines.
As it is part of type expansion, I'll remove the method from this PR.

Reply:
Yes the lookup is performed in ProgramSymboltableBuilder, but I checked the same level condition must also be added in the CreateRedefinesSymbol method, thus smedilol noticed a missing condition.

Reply:
Methods deleted (e4df2af63d783c23ecb150f433a57f208baec0e2). The check on the level is tricky because levels may actually differ. It's the hierarchical level that should be the same. Should we add a new property in Symbol ? Basically HierarchicalLevel = Owner.HierarchicalLevel + 1 and equals 1 for root level-1 dataDef.

Reply:
What is the difference between hierarchical level and level number, if I understand hierarchical level is like the same scope, if so then this is what is checked by CreateRedefinesSymbol , only the same level number check is missing.

Reply:
For example :

```cobol
       01 item.
          05 var1 PIC X.
          04 var2 REDEFINES var1 PIC 9.
```

Different levels (5 vs 4) but same hierarchical level (2).

Reply:
As discussed together, this would be part of another PR focusing on checkers for symbols.